openedx/features/edly/utils.py

```diff
@@ -0,0 +1,255 @@
+import logging
+import re
+
+from courseware.courses import get_course_by_id
+from django.conf import settings
+from openedx.core.djangoapps.site_configuration import helpers as configuration_helpers
+from openedx.features.edly.tasks import send_bulk_mail_to_students
+from student.models import CourseEnrollment
+from xmodule.contentstore.content import StaticContent
+from xmodule.modulestore.django import modulestore
+from xmodule.modulestore.exceptions import InvalidLocationError, ItemNotFoundError
+
+log = logging.getLogger(__name__)
+
+COURSE_OUTLINE_CATEGORIES = ['vertical', 'sequential', 'chapter']
+
+
+def notify_students_about_xblock_changes(xblock, publish, old_content):
+    """
+    This function is responsible for calling the related function by checking the xblock category.
+
+    :param xblock: Block of courses which has been published.
+    :param publish: Param to check if xblock is going to be published.
+    :param old_content: Old data of xblock before updating.
+    """
+    if (publish == 'make_public' and xblock.category in COURSE_OUTLINE_CATEGORIES
+        and not xblock.visible_to_staff_only):
+        _handle_section_publish(xblock)
+    elif xblock.category == 'course_info' and xblock.location.block_id == 'handouts':
+        _handle_handout_changes(xblock, old_content)
+
+
+def get_email_params(xblock):
+    """
+    This function is responsible for generating the email_params which we are
+    going to send to students for any changes in course.
```

Comment:
It might be
`Generates the email params which would be sent to students for any changes in course`

Reply:
Done